s3cortxutils/s3confstore/s3confstore/cortx_s3_confstore.py

```diff
@@ -0,0 +1,134 @@
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# For any questions about this software or licensing,
+# please email opensource@seagate.com or cortx-questions@seagate.com.
+#
+import argparse
+# from cortx.utils.conf_store import Conf
+from cortx.utils.conf_store import ConfStore
+from urllib.parse import urlparse
+import os.path
+import json
+import sys
+
+class S3CortxConfStore:
+  default_index = "default_cortx_s3_confstore_index"
+
+  def __init__(self):
+    """Instantiate constore"""
+    self.conf_store = ConfStore()
+
+  def load_config(self, index: str, backend_url: str):
+    """Load Config into constore"""
+    self.conf_store.load(index, backend_url)
+
+  def get_config(self, index: str, key: str):
+    """Get the key's config from constore"""
+    result_get = self.conf_store.get(index, key, default_val=None)
+    return result_get
+
+  def get_data_config(self, index: str):
+    """Obtains entire config for the given index from constore"""
+    result_get_data = self.conf_store.get_data(index)
+    return result_get_data
+
+  def uttest(self, index: str):
+    """This will be removed to ut directory later"""
+    test_config = {
+      'cluster': {
+        "cluster_id": 'abcd-efgh-ijkl-mnop',
+        "cluster_hosts": 'myhost-1,myhost2,myhost-3',
+      }
+    }
+    with open(r'/tmp/cortx_s3_confstoreuttest.json', 'w+') as file:
+      json.dump(test_config, file, indent=2)
+    self.load_config(index, 'json:///tmp/cortx_s3_confstoreuttest.json')
+
+    result_data = self.get_config(index, 'cluster')
+    if 'cluster_id' not in result_data:
+      print("get_config() failed!")
+      os.remove("/tmp/cortx_s3_confstoreuttest.json")
+      exit(-1)
+
+    result_data = self.get_data_config(index)
+    if 'cluster' not in result_data:
+      print("get_data_config() failed!")
+      os.remove("/tmp/cortx_s3_confstoreuttest.json")
+      exit(-1)
+
+    os.remove("/tmp/cortx_s3_confstoreuttest.json")
+    print("cortx_s3_confstore unit test passed!")
+
+  def run(self):
+    parser = argparse.ArgumentParser(description='Cortx-Utils ConfStore')
+    parser.add_argument("--load", help='Load the backing storage in this index')
+    parser.add_argument("--get", help='Obtain config for this given index using a key, pass --key')
+    parser.add_argument("--get_data", help='Obtains entire config for this given index')
+    parser.add_argument("--key", help='Provide a key to be used in --get')
+    parser.add_argument("--getkey", help='Fetch value of the given key from default index, pass --path to config')
+    parser.add_argument("--path", help='example: JSON:///etc/cortx/myconf.json')
+    parser.add_argument("--uttest", help='An unit test for the load, get and get_data APIs')
+
+    args = parser.parse_args()
+
+    if len(sys.argv) < 2:
+      print("Incorrect args, use -h to review usage")
+      exit(-1)
+
+    if args.path:
```

Comment:
I think we can remove this if block, we are not doing anything here and user will also not send only path option. We need "--path" option only with "--load" or "--getkey"

If you agree, remove it in your subsequent PR. Fine for now.

Reply:
Actually it is possible with current code to send only path option.
# /usr/local/bin/s3confstore --path "JSON:///etc/myconf.json"

My take on this change was, if user provided path, for any types of operation, we should validate it. Please let me know if this needs to be changed. I'll address then in the followup PR.